userguide/go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/google/docsy
+
+go 1.12
```

Comment:
Why is this needed here?

Reply:
If me follow my proposal given in my [comment](https://github.com/google/docsy/pull/871#issuecomment-1036220370) below, I think it's natural to turn the user guide into a hugo module.

Reply:
Removed file `userguide/go.mod`, as requested.